
syntax = 'proto3';

package todo.protobuf;
//import "google/protobuf/empty.proto";



service TodoService{
    rpc  GetAllTodos(EmptyRequest) returns (TodoList);
    rpc  GetTodo(Todo) returns (Todo);
    rpc  CreateTodo(Todo) returns (OperationTodoResponse);
    rpc  DeleteTodo(Todo) returns (OperationTodoResponse);
    rpc  UpdateTodo(Todo) returns (OperationTodoResponse);
}


message EmptyRequest {}

//message KeyValue {
//    string key = 1;
//    string value = 2;
//}
//
//message Items {
////    repeated KeyValue items = 1;
//    map<int, string> items = 2;
//}


message Todo {
   int32 id  = 1;
   string name  = 2;

   enum TODO_STATUS {
       UNKNOWN = 0;
       PENDING  = 1;
       INPROGRESS = 2;
       COMPLETED = 3;
   }
   TODO_STATUS status = 3;
}

message TodoList {
  repeated Todo todos = 1;
}

message OperationTodoResponse {

    enum OP_Status {
        UNKNOWN = 0;
        CREATED  = 1;
        DELETED = 2;
        UPDATED = 3;
        ERROR = 4;
    }

    Todo todo = 1;
    OP_Status op_status = 2;
}
